src/js/index.js

```diff
@@ -1,81 +1,137 @@
-  $menu.addEventListener('click', removeMenu);
+import { $, $id } from './utils/dom.js';
+import LocalStorage from './utils/localStorage.js'
+
+function App () {
+  const MenuListStorage = new LocalStorage('menuList');
+
+  // states
+  // [{name: string; category: string; isSoldOut: boolean;}]
+  let menuList = MenuListStorage.get() ?? [];
+  let category = 'espresso';
+
+  window.onload = () => {
+    initEventListenes();
+    renderAboutMenus();
+  };
+
+  // utils
+  const getMenuId = (target) => {
+    return Number(target.parentElement.dataset.menuId);
+  };
+
+  // events
+  const addMenu = () => {
+    let name = $id('menu-name').value?.trim();
+    if (!name) return;
+
+    menuList = [...menuList, {name, category}];
+    MenuListStorage.set(menuList);
+    renderAboutMenus();
+    $id('menu-name').value = '';
+  };
+
+  const editMenu = ({target}) => {
+    const name = window.prompt('메뉴명을 수정하세요').trim();
+    if (!name) return;
+
+    const menuId = getMenuId(target);
+    menuList = menuList.map((menu, index) => index === menuId ? {...menu, name} : menu);
+    MenuListStorage.set(menuList);
+    renderAboutMenus();
+  };
+
+  const removeMenu = ({target}) => {
+    if (!window.confirm('정말 삭제하시겠습니까?')) return;
+
+    const menuId = getMenuId(target);
+    menuList = menuList.filter((_, index) => index !== menuId);
+    MenuListStorage.set(menuList);
+    renderAboutMenus();
+  };
+
+  const toggleIsSoldOut = ({target}) => {
+    const menuId = getMenuId(target);
+    menuList = menuList.map((menu, index) => 
+      index === menuId ? {...menu, isSoldOut: !menu.isSoldOut} : menu,
+    );
+    MenuListStorage.set(menuList);
+    renderAboutMenus();
+  };
+
+  const setCategory = ({target}) => {
+    const selectedCategory = target.dataset.categoryName;
+    category = selectedCategory;
+    $id('category-title').textContent = `${target.textContent} 메뉴 관리`;
+    renderAboutMenus();
+  };
+
+  // addEventListeners
+  const initEventListenes = () => {
+    const $form = $id('menu-form');
+    $form.addEventListener('submit', (e) => {
+      e.preventDefault();
+      addMenu();
+    });
+    
+    const $menu = $id('menu-list');
+    $menu.addEventListener('click', (e) => {
+      switch (e.target.dataset.action) {
+        case 'edit': 
+          editMenu(e);
+          return;
+        case 'remove':
+          removeMenu(e);
+          return;
+        case 'sold-out':
+          toggleIsSoldOut(e);
+          return;
+      }
+    });
+
+    const $categoryList = $id('cafe-category-name-list');
+    $categoryList.addEventListener('click', setCategory);
+  };
+
+  // renders
+  const renderAboutMenus = () => {
+    const menuListTemplate = menuList
+      .filter(menu => menu.category === category)
+      .map((menu, index) => {
+      return `
+        <li data-menu-id="${index}" class="menu-list-item d-flex items-center py-2">
+          <span class="w-100 pl-2 menu-name${menu.isSoldOut ? ' sold-out' : ''}">${menu.name}</span>
```

Comment:
"sold-out" class 앞의 공백을 삼항연산자 안이 아니라
템플릿 리터럴 안에 넣으면 조금더 가독성에 좋을 것 같아요!
```suggestion
          <span class="w-100 pl-2 menu-name ${menu.isSoldOut ? 'sold-out' : ''}">${menu.name}</span>
```